src/legacy/core_plugins/kibana/public/dashboard/migrations/migrate_to_730_panels.ts

```diff
@@ -0,0 +1,299 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import _ from 'lodash';
+import { i18n } from '@kbn/i18n';
+import semver from 'semver';
+import { DEFAULT_PANEL_WIDTH, DEFAULT_PANEL_HEIGHT } from '../dashboard_constants';
+import {
+  RawSavedDashboardPanelTo60,
+  RawSavedDashboardPanel630,
+  RawSavedDashboardPanel640To720,
+  RawSavedDashboardPanel730ToLatest,
+  RawSavedDashboardPanel610,
+  RawSavedDashboardPanel620,
+} from './types';
+import {
+  SavedDashboardPanelTo60,
+  SavedDashboardPanel620,
+  SavedDashboardPanel630,
+  SavedDashboardPanel610,
+} from '../types';
+
+const PANEL_HEIGHT_SCALE_FACTOR = 5;
+const PANEL_HEIGHT_SCALE_FACTOR_WITH_MARGINS = 4;
+const PANEL_WIDTH_SCALE_FACTOR = 4;
+
+/**
+ * Note!
+ *
+ * The 7.3.0 migrations reference versions that are quite old because for a long time all of this
+ * migration logic was done ad hoc in the code itself, not on the indexed data (migrations didn't even
+ * exist at the point most of that logic was put in place).
+ *
+ * So you could have a dashboard in 7.2.0 that was created in 6.3 and it will have data of a different
+ * shape than some other dashboards that were created more recently.
+ *
+ * Moving forward migrations should be simpler since all 7.3.0+ dashboards should finally have the
+ * same data.
+ */
+
+function isPre61Panel(
+  panel: unknown | RawSavedDashboardPanelTo60
+): panel is RawSavedDashboardPanelTo60 {
+  return (panel as RawSavedDashboardPanelTo60).row !== undefined;
+}
+
+function is61Panel(panel: unknown | RawSavedDashboardPanel610): panel is RawSavedDashboardPanel610 {
+  return semver.satisfies((panel as RawSavedDashboardPanel610).version, '6.1.x');
+}
+
+function is62Panel(panel: unknown | RawSavedDashboardPanel620): panel is RawSavedDashboardPanel620 {
+  return semver.satisfies((panel as RawSavedDashboardPanel620).version, '6.2.x');
+}
+
+function is63Panel(panel: unknown | RawSavedDashboardPanel630): panel is RawSavedDashboardPanel630 {
+  return semver.satisfies((panel as RawSavedDashboardPanel630).version, '6.3.x');
+}
+
+function is640To720Panel(
+  panel: unknown | RawSavedDashboardPanel640To720
+): panel is RawSavedDashboardPanel640To720 {
+  return (
+    semver.satisfies((panel as RawSavedDashboardPanel630).version, '>6.3') &&
+    semver.satisfies((panel as RawSavedDashboardPanel630).version, '<7.3')
+  );
+}
+
+// Migrations required for 6.0 and prior:
+// 1. (6.1) migrate size_x/y/row/col into gridData
+// 2. (6.2) migrate uiState into embeddableConfig
+// 3. (6.3) scale grid dimensions
+// 4. (6.4) remove columns, sort properties
+// 5. (7.3) make sure panelIndex is a string
+function migratePre61PanelToLatest(
+  panel: RawSavedDashboardPanelTo60,
+  version: string,
+  useMargins: boolean,
+  uiState?: { [key: string]: { [key: string]: unknown } }
+): RawSavedDashboardPanel730ToLatest {
+  ['col', 'row'].forEach(key => {
+    if (!_.has(panel, key)) {
```

Comment:
Looks like this could be

```js
if (panel[key] !== undefined) {
```

and then you don't need `lodash` in this module.